### PR TITLE
feature: add out of tree plugin

### PR DESCRIPTION
**Pull request by kakaZhou719**

### CODE REVIEW

File: plugin/plugin.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ const (
 )
 
 const (
+	// Plugin used for golang so file to find the related symbol
+	Plugin             = "Plugin"
+	GolangPlugin       = "GOLANG"
```
Comment (fanux) [inline plugin/plugin.go:38]: The user handles its own plugin type. Don't need add GOLANG type

File: plugin/outoftree_plugin.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+type Golang struct{}
+
+func NewGolangPlugin() Interface {
+	return &Golang{}
+}
+
+func (g Golang) Run(context Context, phase Phase) error {
+	if string(phase) != context.Plugin.Spec.Action || context.Plugin.Spec.Type != GolangPlugin {
```
Comment (fanux) [inline plugin/outoftree_plugin.go:37]: Just load all the .so in plugin dir.

File: plugin/test/post_install.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: sealer.aliyun.com/v1alpha1
+kind: Plugin
+metadata:
+  name: list_nodes.so
+spec:
+  type: GOLANG
```
Comment (fanux) [inline plugin/test/post_install.yaml:20]: The list node plugin type should be LIST_NODE not GOLANG

File: plugin/test/label_nodes.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+			}
+			v.SetLabels(m)
+			v.SetResourceVersion("")
+
+			if _, err := l.client.UpdateNode(&v); err != nil {
+				return fmt.Errorf("current cluster nodes label failed, %v", err)
+			}
+		}
+	}
+	return err
```
Comment (allencloud) [inline plugin/test/label_nodes.go:69]: incorrect, should `return nil` ?

File: plugin/test/label_nodes.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	for _, v := range items {
+		tmps := strings.Split(v, " ")
+		if len(tmps) != 2 {
+			//logger.Warn("label data is no-compliance with the rules! label data: %v", v)
+			continue
+		}
+		ip := tmps[0]
+		labelStr := strings.Split(tmps[1], ",")
+		if len(labelStr) == 0 {
```
Comment (allencloud) [inline plugin/test/label_nodes.go:87]: this will never happen.

File: plugin/plugins.go [hunk 6]
```diff
@@ -74,20 +59,35 @@ func (c *PluginsProcessor) Load() error {
...
 	}
-
 	return nil
 }
 
 func (c *PluginsProcessor) Run(cluster *v1.Cluster, phase Phase) error {
 	var p Interface
 
 	for _, config := range c.Plugins {
+		if ext := filepath.Ext(config.Name); ext == ".so" {
```
Comment (fanux) [inline plugin/plugins.go:78]: Put those logics into a function.

**Review by fanux — APPROVED** [review]

**Pull request merged**